From 6fbf7b593dbd48245fee4411935abb03ba063183 Mon Sep 17 00:00:00 2001
From: Ryan McCabe <rmccabe@redhat.com>
Date: Mon, 16 Jun 2014 10:55:15 -0400
Subject: [PATCH] rgmanager: Fix rg_test exit status on failures

Exit with a non-zero exit code on all failure paths when running
rg_test.

Resolves: rhbz#1033162

Signed-off-by: Ryan McCabe <rmccabe@redhat.com>
---
 rgmanager/src/daemons/test.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/rgmanager/src/daemons/test.c b/rgmanager/src/daemons/test.c
index 1982995..e74a8c1 100644
--- a/rgmanager/src/daemons/test.c
+++ b/rgmanager/src/daemons/test.c
@@ -128,6 +128,7 @@ test_func(int argc, char **argv)
 	conf_setconfig(argv[1]);
        	ccsfd = ccs_lock();
 	if (ccsfd < 0) {
+		ret = -1;
 		printf("Error parsing %s\n", argv[1]);
 		goto out;
 	}
@@ -430,6 +431,7 @@ main(int argc, char **argv)
 		} else {
 			ret = stat(argv[1], &st);
 			if (ret == -1 || !S_ISDIR(st.st_mode)) {
+				ret = -1;
 				break;
 			}
 			fprintf(stderr,
-- 
1.9.3

